liteclient/client.go

```diff
@@ -108,14 +138,29 @@ func PrepareTransaction(inputsBody string, outputsBody string) string {
 		newTransaction.PushOutput(addr, out.Coins, out.Hours)
 	}
 
-	newTransaction.SignInputs(keys)
+	if signatureList == nil {
+		newTransaction.SignInputs(keys)
+	} else {
+		newTransaction.Sigs = make([]cipher.Sig, len(signatureList))
+		for i, sig := range signatureList {
+			decodedSig, e := hex.DecodeString(sig)
+			if e != nil {
+				panic(e)
+			}
+
+			s := cipher.Sig{}
+			if len(decodedSig) != len(s) {
+				log.Panic("Invalid signature length")
+			}
+			copy(s[:], decodedSig[:])
+			newTransaction.Sigs[i] = s
+		}
+	}
```

Comment:
Should this panic if len(sigs) != len(txnInputs)?

Reply:
That is done a few lines below with
```
if err := newTransaction.Verify(); err != nil {
    panic(err)
}
```
`newTransaction.Verify()` checks that in
https://github.com/skycoin/skycoin-lite/blob/67bbce84d181c45321ef57cfb4fcb03cca780198/vendor/github.com/skycoin/skycoin/src/coin/transactions.go#L78